tests/session_test.go

```diff
@@ -78,6 +85,19 @@ func (s *testSession) newClient() *client.Client {
 	return client
 }
 
+func (s *testSession) withUserDB(user string, fn func(client *ent.Client, ctx context.Context)) error {
+	if path, ok := s.userDBPaths[s.userIDs[user]]; ok {
+		if client, err := ent.Open(dialect.SQLite, path); err == nil {
+			fn(client, context.Background())
+			return client.Close()
+		} else {
+			return err
+		}
+	}
+
+	return fmt.Errorf("User not found")
+}
```

Comment:
go has shit error handling so the standard practise that's evolved is to keep the happy path out of `if` blocks and to handle errors in the `if` blocks.

The idiomatic way to write this method would then be:
```suggestion
func (s *testSession) withUserDB(user string, fn func(client *ent.Client, ctx context.Context)) error {
	path, ok := s.userDBPaths[s.userIDs[user]]
	if !ok {
		return fmt.Errorf("user not found")
	}

	client, err := ent.Open(dialect.SQLite, path)
	if err != nil {
		return err
	}

	fn(client, context.Background())

	return client.Close()
}
```

Reply:
Thanks for the suggestion. Fixed.